contracts/factory/src/contract.rs

```diff
@@ -485,9 +487,28 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
             to_binary(&query_pairs(deps, start_after, limit)?)
         }
         QueryMsg::FeeInfo { pair_type } => to_binary(&query_fee_info(deps, pair_type)?),
+        QueryMsg::BlacklistedPairTypes {} => to_binary(&query_blacklisted_pair_types(deps)?),
     }
 }
 
+/// ## Description
+/// Returns a vector that contains blacklisted pair types
+pub fn query_blacklisted_pair_types(deps: Deps) -> StdResult<Vec<PairType>> {
+    let resp = PAIR_CONFIGS
+        .range(deps.storage, None, None, Order::Ascending)
+        .map(|item| {
+            let (_, val) = item.map_err(|_| StdError::generic_err("deserialize error"))?;
+            Ok(val)
+        })
+        .collect::<StdResult<Vec<_>>>()?;
+
+    Ok(resp
+        .into_iter()
+        .filter(|el| el.is_generator_disabled)
```

Comment:
Let's consider `is_disabled` as blacklisting pairs too
```suggestion
        .filter(|el| el.is_disabled || el.is_generator_disabled)
```